### ISSUE TITLE
Bicep is trying to remove virtual network peering

### ISSUE THREAD

**Issue by jamesyoung007** [Azure/arm-template-whatif#278]
## Describe the noise

**Resource type** (i.e. Microsoft.Network/virtualNetworks/virtualNetworkPeerings)

**apiVersion** (i.e. 2019-04-01) 2022-01-01

**Client (PowerShell, Azure CLI, or API)** Azure CLI

**Relevant ARM Template code (we only need the resource object for the above `resourceType` and `apiVersion`, but if it's easier you can include the entire template**

```bicep
resource virtualNetworks_yyy_name_virtualNetworks_yyy_0_name_zzz 'Microsoft.Network/virtualNetworks/virtualNetworkPeerings@2022-01-01' = {
  parent: virtualNetworkszzz_name_resource
  name: '${virtualNetworks_yyy_name}-TO-NZ-SYD-COR-VNT-10.150.112.0'
  properties: {
    peeringState: 'Connected'
    remoteVirtualNetwork: {
      id: virtualNetworks_yyy_externalid
    }
    allowVirtualNetworkAccess: true
    allowForwardedTraffic: true
    allowGatewayTransit: false
    doNotVerifyRemoteGateways: false
    peeringSyncLevel: 'FullyInSync'
    remoteAddressSpace: {
      addressPrefixes: [
        '10.0.2.0/24'
      ]
    }
    remoteVirtualNetworkAddressSpace: {
      addressPrefixes: [
        '10.0.2.0/24'
      ]
    }
    useRemoteGateways: true

  }
}
```

**Expected response (i.e. "I expected no noise since the template has not been modified since the resources were deployed)**
It should not be displayed if there is no change

**Current (noisy) response (either include a screenshot of the what-if output, or copy/paste the text)**

 ~ Microsoft.Network/virtualNetworks/xxx [2022-01-01]
    ~ properties.virtualNetworkPeerings: [
      ~ 0:

        - properties.peerCompleteVnets: true
        - properties.remoteGateways: [
            0:

              id: "/subscriptions/xxx/resourceGroups/NZ-SYD-COR-ARG-Network/providers/Microsoft.Network/virtualNetworkGateways/zzz"

          ]

**Additional context**
Some times, it will report error message when we apply it. 
But more often, it will report no errors, shows the deployment was successful. 
Hence, it can be a bit confusing.

**Comment by aolszowka** [comment]
FWIW We see something similar with our two VNET's:

```
~ properties.virtualNetworkPeerings: [
      ~ 0:

        - properties.peerCompleteVnets: true

      ~ 1:

        - properties.peerCompleteVnets: true

      ]
```

There is no documentation I can find for this property (`virtualNetworkPeerings`) online. In fact, this GitHub issue is the only reference I was able to find to it (I had to use Bing, Google had NOTHING). I have not attempted a deployment due to concerns that this will break something existing.

My best guess is that it is some private variable that is controlled by `peeringSyncLevel`, in our case the VNETs are set to `FullyInSync`, My guess is that `peerCompleteVnets` might have been an old property/implementation detail that was replaced by `peeringSyncLevel` in API `2021-02-01` (see https://learn.microsoft.com/en-us/azure/templates/microsoft.network/change-log/virtualnetworks) but never cleaned up. But that is complete speculation, don't bet the farm on it.

@jamesyoung007 were you able to figure anything out related to this?

@alex-frankel I see you edited this; any chance you could point us in the right direction? I also don't mind opening a support case to route it that way as well.

**Comment by jamesyoung007** [comment]
I was advised that this is a read only property. And I tried, it didn't make any change.
However, I have been told with another value is read only, it turned out to be not true.
(Even worse, in some subscriptions, it's read only, in other subscriptions, it can be modified)
I suggest you test it out.

**Comment by aolszowka** [comment]
Closing the loop from our end on this; hopefully this helps some other DevOps engineer out there:

I had reached out to Azure Support and got a non-committal answer as to what it affects exactly. I went ahead and ran it in a lower valued environment and the changes had no affect that I can discern, which really lends credence to the fact that this is indeed noise.

FWIW We were also seeing something similar complaining about `properties.networkIntentPolicies` with regards to the subnet we have setup for Azure SQL Managed Instance; again, on a different support case <strike>really didn't get the answer we were looking for, but at least for us it did nothing</strike> See Edit 2022/12/08 below:

```
- properties.networkIntentPolicies: [
            0:
              id: "/subscriptions/SUBSCRIPTION-ID/resourceGroups/RESOURCE-GROUP/providers/Microsoft.Network/networkIntentPolicies/mi_default_REDACTED-GUID-XXX-XX-X-XX"
          ]
```

Reruns also still show this, again more noise.

This feature is incredibly powerful, hopefully our reports can work towards limiting the noise out of it.

**Edit 2022/12/08**
On the `networkIntentPolicies` issue we heard back, this is caused by delegation of the subnet to `Microsoft.Sql/managedInstances` as per this snippet:

```
"delegations": [
  {
    "name": "miDelegation",
    "id": "[resourceId('Microsoft.Network/virtualNetworks/subnets/delegations', variables('virtualNetworkName'), 'ManagedInstance', 'miDelegation')]",
    "properties": {
        "serviceName": "Microsoft.Sql/managedInstances"
    },
    "type": "Microsoft.Network/virtualNetworks/subnets/delegations"
  }
]                                
```

According to support this causes the `networkIntentPolicies` to be modified to have this intent, but `--what-if` is not smart enough to understand that this will be applied apparently hence the noise.

This indicates probably more noise in the system I will see if I have the energy to open a new case that details it.

Opened https://github.com/MicrosoftDocs/azure-docs/issues/102527 for the networkIntetPolicies issue.
